src/NotEmpty.php

```diff
@@ -100,19 +100,8 @@ public function __construct($options = null)
         }
 
         if (is_array($options)) {
-            if (!array_key_exists('type', $options)) {
-                $detected = 0;
-                $found    = false;
-                foreach ($options as $option) {
-                    if (in_array($option, $this->constants, true)) {
-                        $found = true;
-                        $detected += array_search($option, $this->constants);
-                    }
-                }
-
-                if ($found) {
-                    $options['type'] = $detected;
-                }
+            if (($type = $this->calculateTypeValue($options)) != 0) {
+                $options['type'] = $type;
```

Comment:
It's better create a new $options array here for to clean up from those old values are now represented by $type.